### PR TITLE
Rewrite the mentoring docs based on SR2021 experiences

### PR DESCRIPTION
**Pull request by PeterJCLaw**
This updates this document to reflect the approximate process used during the SR2021 cycle. This is broadly similar to the previous approach, though is much more open (notably using by default Slack rather than person-to-person emails) and supports remote mentoring.

This document now also contains much more information intended for the (potential) mentors rather than focussing entirely on the coordinator.

Fixes https://github.com/srobo/runbook/issues/99

See also https://github.com/srobo/volunteer-emails/pull/43 which adds the example email which this PR is removing.

### CODE REVIEW

**Comment by PeterJCLaw** [comment]
@sedders123 and @antoinepetty in particular -- I'd be interested to hear about your experiences as mentors this year and whether I've captured here everything that you needed/wanted/found was useful/etc. as well as a general review.

File: docs/teams/mentoring.md [hunk 1]
```diff
@@ -1,68 +1,174 @@
...
+As well as providing support and guidance to the team, this volunteer typically
+becomes the competitors' main contact with Studdent Robotics.
+
+For this role, mentors need good access to information which may be pertinent to
+competitors and compliment the information we send directly to team leaders.
+
+This page provides an overview of how we organise mentoring and is intended both
+for potential mentors and a team coordinator looking to organise mentoring.
+
+## What mentoring entails
```
Comment (PeterJCLaw) [inline docs/teams/mentoring.md:13]: Do we want to also add something here about the mentor being another person to bounce ideas off? (and one who is perhaps slightly detached from the team's own thinking) Building a rapport with the team is definitely an important part of mentoring, so perhaps.

File: docs/teams/mentoring.md [hunk 2]
```diff
@@ -1,68 +1,174 @@
...
-    * Confirm over 18
-    * Name
-    * Email address
-    * Location
-    * How far willing to travel to mentor
-    * Roughly how often could commit to visiting a team
-    * What skills do you possess making you suitable mentor
-    * Previous SR experience
-
-## Example of email sent to potential Mentors asking for volunteers in 2017
```
Comment (RealOrangeOne) [inline docs/teams/mentoring.md:19]: Thoughts on capturing this email content somewhere?

File: docs/teams/mentoring.md [hunk 3]
```diff
@@ -1,68 +1,174 @@
----
-original:
-  authors: Fiona Gillan
-  url: https://github.com/fgillan/sr-team-coord/wiki/Mentoring
----
 # Mentoring
```
Comment (RealOrangeOne) [inline docs/teams/mentoring.md:1]: A lot of this new content is super useful, although it feels slightly duplicated of https://studentrobotics.org/runbook/volunteering/mentor-guidance/. Perhaps worth combining the pages and cross-referencing?

File: docs/teams/mentoring.md (hunk 2 above)
Comment (PeterJCLaw) [reply docs/teams/mentoring.md:19]: Yes: https://github.com/srobo/volunteer-emails/pull/43 does this.

File: docs/teams/mentoring.md (hunk 3 above)
Comment (PeterJCLaw) [reply docs/teams/mentoring.md:1]: My goal with this page is to capture the "what" of mentoring, as a guide for organising it as a whole and for people who haven't done it before (including people who haven't heard of it before). I think the focus of the other one is much more "how".
That said, I haven't re-reviewed the other one in detail to compare the overlap. I suspect that moving the other one into this teams folder could be a useful thing to do, but deliberately didn't change that here.

File: docs/teams/mentoring.md (hunk 1 above)
Comment (RealOrangeOne) [reply docs/teams/mentoring.md:13]: I think we should. It's quite an important detail that a mentor not be "just another team member"

Comment (PeterJCLaw) [reply docs/teams/mentoring.md:13]: 9dc0230. Not entirely happy with the framing though as I bet a lot of people might read that and think "I don't know how to do that".

**Review by RealOrangeOne — APPROVED** [review]

**Pull request merged**